ztp/policygenerator/policyGen/policyBuilder.go

```diff
@@ -246,6 +256,18 @@ func (pbuilder *PolicyBuilder) setValues(sourceMap map[string]interface{}, value
 			sourceMap[k] = valueMap[k]
 		}
 	}
+
+	// If the user provides values in the PGT spec/data sections for which there
+	// is not a corresponding key in the source-cr we need to simply copy over
+	// the user supplied values
+	for k, v := range valueMap {
+		if sourceMap[k] != nil {
```

Comment:
it will be more clean as 
`if sourceMap[k] == nil {
  sourceMap[k] = v
}
`

Reply:
:+1: